tardis/utilities/executors/sshexecutor.py

```diff
@@ -3,20 +3,51 @@
 from ...interfaces.executor import Executor
 from ..attributedict import AttributeDict
 
+import asyncio
 import asyncssh
 
 
 @enable_yaml_load("!SSHExecutor")
 class SSHExecutor(Executor):
     def __init__(self, **parameters):
         self._parameters = parameters
+        self._ssh_connection = None
+        self._lock = None
+
+    async def establish_connection(self):
```

Comment:
Should not be part of the public interface.

```suggestion
    async def _establish_connection(self):
```